hivemind/utils/grpc.py

```diff
@@ -215,9 +215,17 @@ def serialize_torch_tensor(tensor: torch.Tensor, compression_type=CompressionTyp
 
 def deserialize_torch_tensor(serialized_tensor: runtime_pb2.Tensor) -> torch.Tensor:
     # TODO avoid copying the array (need to silence pytorch warning, because array is not writable)
+
+    def construct_torch_tensor(serialized_tensor, array, dtype=None):
```

Comment:
1. Better to move the function from outer scope
2. Better to make `array` the first argument and to pass only the tensor size as input